rename/rename.sh

```diff
@@ -1,9 +1,8 @@
 #!/usr/bin/env bash
 
 mvformat() {
-    test -n "$1" || return 1
-    for i in ./*; do
-        NAME=$(echo $i | egrep -o '.*\.')
-        mv "$i" "$NAME$1"
+    [ "$#" -eq 1 ] || return 1
+    for i in *; do
+	mv "$i" "${i%.*}.$1"
```

Comment:
Just a touch more traditional. The `$#` is a special variable assigned the value of the number of positional parameters sent to the script, or function if this is used within one.

The whole NAME thing wasn't needed, so this is a lot cleaner and avoids needing `grep`. Bash is capable of doing plenty in its own stead, such as matching regular expressions.

I would usually do these kind of operations on the fly though, as it tends to just be a case of:

```
for FILE in *.mp3; { mv "$FILE" "${FILE%.*}.ogg"; }
```

My favorite usage of such trickery is when converting files with `ffmpeg`. :D